api/buckets/service.go

```diff
@@ -1606,33 +1607,87 @@ func (s *Service) Archive(ctx context.Context, req *pb.ArchiveRequest) (*pb.Arch
 		return nil, fmt.Errorf("parsing cid path: %s", err)
 	}
 
-	ffsi, err := s.Collections.FFSInstances.Get(ctx, req.GetKey())
-	if err != nil {
-		return nil, fmt.Errorf("getting ffs instance data: %s", err)
+	var ffsInfo *mdb.FFSInfo
+	var owner crypto.PubKey
+	var isAccount bool
+	if acct := accountFromContext(ctx); acct != nil {
+		ffsInfo = acct.FFSInfo
+		owner = acct.Key
+		isAccount = true
+	} else if user := userFromContext(ctx); user != nil {
+		ffsInfo = user.FFSInfo
+		owner = user.Key
+		isAccount = false
+	}
+
+	createNewFFS := func() error {
+		id, token, err := s.PGClient.FFS.Create(ctx)
+		if err != nil {
+			return fmt.Errorf("creating new ffs instance: %v", err)
+		}
+		if isAccount {
+			_, err = s.Collections.Accounts.UpdateFFSInfo(ctx, owner, &mdb.FFSInfo{ID: id, Token: token})
+		} else {
+			_, err = s.Collections.Users.UpdateFFSInfo(ctx, owner, &mdb.FFSInfo{ID: id, Token: token})
+		}
+		if err != nil {
+			return fmt.Errorf("updating user/account with new ffs information: %v", err)
+		}
+		return nil
+	}
+
+	tryAgain := fmt.Errorf("new ffs instance created, please try again in 30 seconds to allow time for wallet funding")
+
+	// case where account/user was created before bucket archives were enabled.
+	// create a ffs instance for them.
+	if ffsInfo == nil {
+		if err := createNewFFS(); err != nil {
+			return nil, err
+		}
+		return nil, tryAgain
```

Comment:
If we don't find any FFS info for the caller, this means their record was created before this code was deployed, so we should lazily create the FFS instance for them.